cmd/frontend/graphqlbackend/textsearch.go

```diff
@@ -599,14 +599,14 @@ func searchFilesInRepos(ctx context.Context, args *search.Args) (res []*FileMatc
 			}
 
 			args := *args
-			if args.Pattern.IsStructuralPat && searcherReposFilteredFiles != nil {
+			if args.PatternInfo.IsStructuralPat && searcherReposFilteredFiles != nil {
 				// Modify the search query to only run for the filtered files
 				if v, ok := searcherReposFilteredFiles[string(repoRev.Repo.Name)]; ok {
-					patternCopy := *args.Pattern
-					args.Pattern = &patternCopy
-					includePatternsCopy := make([]string, len(args.Pattern.IncludePatterns))
-					copy(includePatternsCopy, args.Pattern.IncludePatterns)
-					args.Pattern.IncludePatterns = append(includePatternsCopy, v...)
+					patternCopy := *args.PatternInfo
```

Comment:
nit:
```suggestion
					tmp := *args.PatternInfo
```

I believe `tmp` is sufficient because its scope is only two lines (`patternInfoCopy` feels unnecessarily verbose).

Reply:
I take your point, though I'd like this PR to be a completely mechanical change. These changes are to set up easier refactoring later and to search over everything called, e.g., `lower(patternInfo)`. #7183 is also quite verbose and we can cut down verbosity/renaming after refactoring or as a separate diff.